server/src/server.ts

```diff
@@ -0,0 +1,258 @@
+/**
+ * AGLint Language Server for VSCode
+ */
+
+import {
+    createConnection,
+    TextDocuments,
+    Diagnostic,
+    DiagnosticSeverity,
+    ProposedFeatures,
+    InitializeParams,
+    DidChangeConfigurationNotification,
+    InitializeResult,
+} from 'vscode-languageserver/node';
+
+import { TextDocument } from 'vscode-languageserver-textdocument';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {
+    LinterConfig, scan, walk, Linter, buildConfigForDirectory,
+} from '@adguard/aglint';
+import { ParsedPath, join as joinPath } from 'path';
+import { fileURLToPath } from 'url';
+
+// Create a connection for the server, using Node's IPC as a transport.
+// Also include all preview / proposed LSP features.
+const connection = createConnection(ProposedFeatures.all);
+
+// Create a simple text document manager.
+const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
+
+/**
+ * URL to the AGLint repository
+ */
+const AGLINT_URL = 'https://github.com/AdguardTeam/AGLint';
+
+let hasConfigurationCapability = false;
+let hasWorkspaceFolderCapability = false;
+
+/**
+ * Root folder of the VSCode workspace
+ */
+let workspaceRoot: string | undefined;
+
+type CachedPaths = { [key: string]: LinterConfig };
+
+/**
+ * Cache of the scanned workspace
+ */
+let cachedPaths: CachedPaths | undefined;
+
+/**
+ * Scan the workspace and cache the result.
+ */
+async function cachePaths(): Promise<boolean> {
+    // Cache the scan result
+    if (workspaceRoot) {
+        try {
+            // Get the config for the cwd, should exist
+            const rootConfig = await buildConfigForDirectory(workspaceRoot);
+
+            const scanResult = await scan(workspaceRoot);
+
+            // Create a map of paths to configs
+            const newCache: CachedPaths = {};
+
+            await walk(
+                scanResult,
+                {
+                    file: async (path: ParsedPath, config: LinterConfig) => {
+                        const filePath = joinPath(path.dir, path.base);
+
+                        // Add the file path to the new cache map with the resolved config
+                        newCache[filePath] = { ...config };
+                    },
+                },
+                rootConfig,
+            );
+
+            // Update the whole cache
+            cachedPaths = { ...newCache };
+
+            connection.console.info(`AGLint successfully scanned and cached the workspace: ${workspaceRoot}`);
```

Comment:
Logging configuration files that have been used will also be useful

Reply:
Ok, but could it be a problem that this sometimes gives long output?

Reply:
Just the file names

Reply:
Future TODO, I opened an issue for it